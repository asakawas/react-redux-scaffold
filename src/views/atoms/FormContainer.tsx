import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import { spacing } from '@material-ui/system';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';

const SpacingContainer = styled(Container)`
  ${spacing}
  justify-content: center;
  display: flex;
  min-width: 30em;
  text-align: left;
`;

const FormContainer: FunctionComponent = ({ children }) => (
  <SpacingContainer maxWidth={false} component={Paper} p={2}>
    {children}
  </SpacingContainer>
);

export default FormContainer;
