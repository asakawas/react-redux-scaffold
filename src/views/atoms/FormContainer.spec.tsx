import React from 'react';
import { render } from '@testing-library/react';
import FormContainer from './FormContainer';

describe('FormContainer', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(
      <FormContainer>
        <div />
      </FormContainer>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
