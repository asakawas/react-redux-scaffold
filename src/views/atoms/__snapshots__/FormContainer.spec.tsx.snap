// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  padding: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 30em;
  text-align: left;
}

<div
    class="MuiPaper-root MuiContainer-root c0 MuiPaper-elevation1 MuiPaper-rounded"
    p="2"
  >
    <div />
  </div>
</DocumentFragment>
`;
